tests/typowalking/wordwalker.js

```diff
@@ -27,7 +28,8 @@ bender.test( {
 		wordwalker = new editor.plugins.nanospell.WordWalker(range);
 
 		while (currWordObj = wordwalker.getNextWord()) {
-			wordsReturned.push(currWordObj.word);
+			word = currWordObj.word;
+			if (word) wordsReturned.push(word);
```

Comment:
there may be some validity to eventually setting up tests to test the word ranges returned, not just the text of the words, as they are separate object properties.  In the course of troubleshooting this issue, I ran across several cases where the returned word range did not correspond to the returned text, which I'm sure caused some of the observed issues.

The separation of words for the AJAX call at the beginning of spellcheck relies on the actual `word` text while the insertion of spellcheck spans relies on the `wordRange` property, so it makes sense to test both.